compiler/src/iree/compiler/Dialect/Flow/Transforms/DispatchWithTransformDialect.cpp

```diff
@@ -26,6 +30,33 @@ namespace iree_compiler {
 namespace IREE {
 namespace Flow {
 
+namespace {
+
+/// Builds transform IR forming dispatch regions for reductions.
+void buildReductionDispatch(ImplicitLocOpBuilder &builder, Value scopeH) {
+  auto pdlOperation = pdl::OperationType::get(builder.getContext());
+  SmallVector<Type> matchedTypes(4, pdlOperation);
+  auto matched = builder.create<transform_ext::MatchCallbackOp>(
+      matchedTypes, "reduction_partial",
+      transform::FailurePropagationMode::Suppress, scopeH);
+
+  auto takeFirst = builder.create<transform_ext::TakeFirstOp>(
```

Comment:
Please use:
```
std::pair<Value, Value> mlir::iree_compiler::buildSelectFirstNonEmpty(
    ImplicitLocOpBuilder &b, Value handle1, Value handle2) {
```

Reply:
This would make Flow depend on Codegen, I suppose this is not desirable from the layering perspective.

Reply:
I'll move that to iree-dialects.